sql/planbuilder/scope.go

```diff
@@ -82,26 +99,47 @@ func (s *scope) resolveColumn(table, col string, checkParent bool) (scopeColumn,
 	return c, true
 }
 
+// triggerCol is used to hallucinate a new column during trigger DDL
+// when we fail a resolveColumn.
+func (s *scope) triggerCol(table, col string) (scopeColumn, bool) {
+	// hallucinate tablecol
+	for _, t := range s.b.TriggerCtx().UnresolvedTables {
+		if strings.EqualFold(t, table) {
+			col := scopeColumn{table: t, col: col}
+			id := s.newColumn(col)
+			col.id = id
+			return col, true
+		}
+	}
+	if table == "" {
+		col := scopeColumn{col: col}
+		id := s.newColumn(col)
+		col.id = id
+		return col, true
+	}
+	return scopeColumn{}, false
+}
+
 // getExpr returns a columnId if the given expression has
 // been built.
-func (s *scope) getExpr(name string) (columnId, bool) {
+func (s *scope) getExpr(name string, checkCte bool) (columnId, bool) {
```

Comment:
Probably better to have a getExprInCte method than add this extra param

Reply:
There can be CTEs in higher scopes whose naming precedence is interleaved